Doc/tutorial/errors.rst

```diff
@@ -147,10 +147,51 @@ For example, the following code will print B, C, D in that order::
 Note that if the *except clauses* were reversed (with ``except B`` first), it
 would have printed B, B, B --- the first matching *except clause* is triggered.
 
-All exceptions inherit from :exc:`BaseException`, and so it can be used to serve
-as a wildcard. Use this with extreme caution, since it is easy to mask a real
-programming error in this way!  It can also be used to print an error message and
-then re-raise the exception (allowing a caller to handle the exception as well)::
+When an exception occurs, it may have associated values, also known as the
+exception's *arguments*. The presence and types of the arguments depend on the
+exception type.
+
+The *except clause* may specify a variable after the exception name.  The
+variable is bound to the exception instance which typically has an ``args``
+attribute that stores the arguments. For convenience, builtin exception
+types define :meth:`__str__` to print all the arguments without explicitly
+accessing ``.args``.  ::
+
+   >>> try:
+   ...     raise Exception('spam', 'eggs')
+   ... except Exception as inst:
+   ...     print(type(inst))    # the exception instance
+   ...     print(inst.args)     # arguments stored in .args
+   ...     print(inst)          # __str__ allows args to be printed directly,
+   ...                          # but may be overridden in exception subclasses
+   ...     x, y = inst.args     # unpack args
+   ...     print('x =', x)
+   ...     print('y =', y)
+   ...
+   <class 'Exception'>
+   ('spam', 'eggs')
+   ('spam', 'eggs')
+   x = spam
+   y = eggs
+
+The exception's :meth:`__str__` output is printed as the last part ('detail')
+of the message for unhandled exceptions.
+
+:exc:`BaseException` is the common base class of all exceptions, including
+:exc:`Exception` which is a subclass of all exceptions apart a few which
+are not typically handled, because they are used to indicate that the
+program should terminate. They include :exc:`SystemExit` which is raised by
```

Comment:
"apart" should be "apart from", but the whole sentence is rather complex. Here's a suggested rewrite:

```suggestion
:exc:`Exception` is the common base class of almost all exceptions. It is
itself a subclass of :exc:`BaseException`. A few other exceptions also
derive directly from :exc:`BaseException`, because they indicate that the
program should terminate and should not typically be handled. These
include :exc:`SystemExit` which is raised by
```

This avoids overly long sentences and gives prominence to `Exception`, which I think is more important to cover in the tutorial.

Reply:
I don't object to rewording this sentence, but I don't think the suggested change works.  This doesn't really make sense at this point:

"A few other exceptions also derive directly from :exc:`BaseException`, because they indicate that the program should terminate and should not typically be handled."

because it's not clear why the fact that "they indicate .. and should not be handled" explain why they are not BaseException subclasses.

Reply:
Good point, reworded my suggestion above

Reply:
That's better, but we are no longer saying that BaseException is the base class of all exceptions.

Reply:
True, but do we need to? This is the tutorial, not the language reference.

Reply:
I don't think it's clear why BaseException exists without that point.

Reply:
I've reworded this paragraph, hopefully the new version is better.